### PR TITLE
[NU-1291] Make processing type to category mapping 1-to-1 instead of 1-to-many

### PR DESCRIPTION
**Pull request by arkadius**
## Checklist before merge
- [x] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [x] Parts of the code that are not easy to understand are documented in the code
- [x] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [x] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [x] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Review by pielas — CHANGES_REQUESTED** [review]
Just a few small tweaks

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 1]
```diff
@@ -40,37 +40,70 @@ class ComponentResourcesApiSpec
...
-        //        "streaming-enricher-env",
-        //        "streaming-customnode-for-each",
-        //        "streaming-source-request",
-        //        "streaming-sink-response",
-        //        "streaming-customnode-union",
-        "streaming-source-boundedsource",
-        "switch",
-        "streaming-source-classinstancesource",
-        "streaming-customnode-enrichwithadditionaldata",
+      val correctListForTestUser: List[String] = List(
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:43]: Sorted list is easier to use for verification where are differences and to change "expected" set when something is changed in our development set of components.

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 2]
```diff
@@ -85,78 +118,135 @@ class ComponentResourcesApiSpec
             .statusCode(200)
             .extractToStringsList("id")
 
-        componentIdListForTestUser should contain theSameElementsAs correctListForTestUser
+        componentIdListForTestUser.sorted should contain theSameElementsAs correctListForTestUser
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:121]: see comment above

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessCategoryService.scala [hunk 3]
```diff
@@ -108,16 +130,22 @@ object ConfigProcessCategoryService extends LazyLogging {
         s"Ambiguous category to scenario type mapping detected for categories: $ambiguousCategoryToProcessingTypeMappings"
       )
 
+  private[process] final case class ProcessingTypeToCategoryMappingAmbiguousException(
+      ambiguousProcessingTypeToCategoryMappings: Map[ProcessingType, Set[Category]]
+  ) extends IllegalStateException(
+        s"Ambiguous scenario type to category mapping detected for categories: $ambiguousProcessingTypeToCategoryMappings"
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessCategoryService.scala:136]: Basically our users have a very few processing types (mostly one). One of our biggest clients has it around ~8 and I suspect that only one processing type will be printed for them. Taking this into consideration, are you still convinced to print keys in separate lines?

[the remark this answers is not in the document]
Comment (pielas) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessCategoryService.scala]: In this case it could be probably left as it is

**Pull request merged**